src/ProductConstructionService/ProductConstructionService.Api/Controllers/StatusController.cs

```diff
@@ -45,8 +45,20 @@ public async Task<IActionResult> StopPcsWorkItemProcessors()
     {
         return Ok(await PerformActionOnAllProcessors(async stateCache =>
         {
-            await stateCache.SetStateAsync(WorkItemProcessorState.Stopping);
-            return (stateCache.ReplicaName, WorkItemProcessorState.Stopping);
+            var state = await stateCache.GetStateAsync();
+            switch (state)
+            {
+                case WorkItemProcessorState.Stopping:
+                case WorkItemProcessorState.Working:
+                    await stateCache.SetStateAsync(WorkItemProcessorState.Stopping);
+                    return (stateCache.ReplicaName, WorkItemProcessorState.Stopping);
+                case WorkItemProcessorState.Initializing:
+                    throw new Exception("Can't stop the service while initializing, try again later");
```

Comment:
It'd be better to return bad request rather than 500

Reply:
Good point, think this is the appropriate exception type